### ISSUE TITLE
Staking logic in the `x/dualstaking` module

### ISSUE THREAD

**Issue by lukasz-zimnoch** [mezo-org/mezod#125]
Depends on: https://github.com/thesis/mezo/issues/124

This task is about implementing the actual logic of staking within the new `x/dualstaking` module. It focuses on staking logic and should abstract away the complexity of state management using Go interfaces. The actual state management will be handled as part of the follow-up task #126.

The detailed scope of this ticket is as follows:

1. Implement the `x/dualstaking` message server. The server should accept messages allowing to:
    - Stake BTC/MEZO and receive veBTC/veMEZO in return
    - Delegate veBTC/veMEZO to a validator (self-delegation should be possible as well)
    - Stake + delegate at once (let's call it auto-delegation for the purpose of this task)
2. Implement the `x/dualstaking` query server. For now, the query server should expose information about existing stakes and delegations.
3. Implement a staking mechanism that will:
    - Accept BTC/MEZO from the stake owner as input
    - Escrow received BTC/MEZO under `x/dualstaking` module account (`x/bank` transfer)
    - Mint appropriate amount of veBTC/veMEZO in the `x/bank` module
    - In case of auto-delegation, trigger delegation logic. Otherwise, transfer minted veBTC/veMEZO to the owner
4. Implement a delegation mechanism that will:
    - Accept veBTC/veMEZO as input 
    - ~Escrow received veBTC/veMEZO under `x/dualstaking` module's account (`x/bank` transfer)~ **Delegated veBTC/veMEZO should stay on owner's account (needs final confirmation)**
    - Calculate how many Synthetic Staking Coin (SSC) should be minted to produce the proper validation power value for the given validator (as showcased in [Dual staking (core): Validation Power Example](https://coda.io/d/Mezo_d_vIDNPK008/Product-Pitch-Drafts_su1QZ#Draft-Product-Pitches_tuH_B/r29)). Note that this may imply a need to rebase SSC balances for all validators.
    - Mint the calculated amount of SSC to the validator
    - Call `x/staking` module to stake and delegate minted SSC for the validator

The delegation mechanism (point 4) is the most challenging and problematic. There is also a concern related to performance as rebasing balances for all validators on every block may be heavy. Let's start by prototyping this mechanism and covering it with some benchmark tests (e.g. for 100, 1000, and 10000 validators). The crucial thing here is that the prototype must interact with a real storage that stores data on disk (this is the most likely bottleneck here). The real storage can be the module's keeper or a custom mechanism, whichever will be easier to use here. Note that if we decide to use the keeper, we will need to implement part of the state management covered by #126. 

⚠️ **Important note: Remember to ensure that `x/dualstaking` does not handle Cosmos transactions directly. The message server of this module will be called by the Staking precompile (#123) which should be the only entrypoint here.**
